pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -99,6 +99,55 @@ type oldXfrmStateKey struct {
 	Dst [16]byte
 }
 
+type IPSecParameters struct {
+	// The BootID for the local host, used for bookkeeping.
+	LocalBootID string
+	// The BootID for the remote host, used for bookkeeping.
+	RemoteBootID string
+	// The direction of the created XFRM policy.
+	Dir IPSecDir
+	// The source subnet selector for the XFRM policy/state
+	SourceSubnet *net.IPNet
+	// The destination subnet selector for the XFRM policy/state
+	DestSubnet *net.IPNet
+	// The source security gateway IP used to define an IPsec tunnel mode SA
+	// For OUT policies this is the resulting source address of an ESP encrypted
+	// packet.
+	// For IN/FWD this should identify the source SA address of the state which
+	// decrypted the the packet.
+	SourceTunnelIP *net.IP
+	// The destination security gateway IP used to define an IPsec tunnel mode SA
+	// For OUT policies this is the resulting destination address of an ESP encrypted
+	// packet.
+	// For IN/FWD this should identify the destination SA address of the state which
+	// decrypted the the packet.
+	DestTunnelIP *net.IP
+	// The ReqID used for the resulting XFRM policy/state
+	ReqID int
```

Comment:
IMO, these parameters should more or less plainly spell out how the resulting policy/state looks. I don't want to have hidden resulting in the policy which can't be seen from the high level structure. 

But even more-so, in a future update, we will be removing the encrypted overlay distinction all together. It's kinda a hack which we have a solution for. Happy to talk more in detail about this in real time if you'd like.

Reply:
Sure. I only commented this because you wrote:
> Having a well defined structure allows us to declare exactly how an XFRM
policy should look at a higher level

But I don't mind either way :smiley_cat: